camddmw/procedures/create_partitions.sql

```diff
@@ -1,262 +1,290 @@
-		RAISE NOTICE '%', cmdStmt;
-		IF executeFlag = 'Y' THEN
-			EXECUTE cmdStmt;
-			RAISE NOTICE '-- executed --';
-		END IF;
-	END LOOP;
-
--------------------------------------------------------------------------------------------------------------------
-	tableName := 'unit_fact';
-	partitionName := tableName || '_unit_fact_tmp_p';
-	EXECUTE format('
-		SELECT MAX(CAST(REPLACE(table_name, %L, %L) AS integer))
-		FROM information_schema.tables
-		WHERE table_name like ''%s%%'';
-	', partitionName, '', partitionName) INTO partitionNumber;
-	cmdStmt := format('CREATE TABLE IF NOT EXISTS %s.%s%s PARTITION OF %s.%s
-				    FOR VALUES FROM (%L) TO (%L);', schemaName, partitionName, partitionNumber+1, schemaName, tableName, year, year+1);
-	RAISE NOTICE '%', cmdStmt;
-	IF executeFlag = 'Y' THEN
-		EXECUTE cmdStmt;
-		RAISE NOTICE '-- executed --';
-	END IF;
-
--------------------------------------------------------------------------------------------------------------------
-	tableName := 'unit_type_year_dim';
-	partitionName := tableName || '_unit_type_yr_dim_p';
-	EXECUTE format('
-		SELECT MAX(CAST(REPLACE(table_name, %L, %L) AS integer))
-		FROM information_schema.tables
-		WHERE table_name like ''%s%%'';
-	', partitionName, '', partitionName) INTO partitionNumber;
-	cmdStmt := format('CREATE TABLE IF NOT EXISTS %s.%s%s PARTITION OF %s.%s
-				    FOR VALUES FROM (%L) TO (%L);', schemaName, partitionName, partitionNumber+1, schemaName, tableName, year, year+1);
-	RAISE NOTICE '%', cmdStmt;
-	IF executeFlag = 'Y' THEN
-		EXECUTE cmdStmt;
-		RAISE NOTICE '-- executed --';
-	END IF;
-
-END
-$BODY$;
+CREATE OR REPLACE PROCEDURE camddmw.create_partitions(
+	IN year integer,
+	IN executeflag character DEFAULT 'N'::bpchar)
+LANGUAGE 'plpgsql'
+AS $BODY$
+
+DECLARE
+	counter integer;
+	toDate date;
+	fromDate date;
+	cmdStmt text;
+	tableName text;
+	partitionName text;
+	partitionNumber integer;
+	partitionExpression text;
+	schemaName text := 'camddmw';
+BEGIN
+	IF executeFlag = 'N' THEN
+		RAISE NOTICE 'The executeFlag is set to ''N'' so partitions will not be created. DRY RUN ONLY! Set executeFlag to ''Y'' to create partitions.';
+	END IF;	
+
+	--NEED TO DISCUSS HOW FAR OUT TO CREATE THESE
+	--camddmw.allowance_holding_dim
+	--CONSTRAINT allowance_holding_dim_all_hold_dim_p66_pkey PRIMARY KEY (vintage_year, prg_code, start_block)
+	--FOR VALUES FROM ('2060') TO ('2061')
+
+-------------------------------------------------------------------------------------------------------------------
+	tableName := schemaName || '.annual_unit_data';
+	cmdStmt := format('CREATE TABLE IF NOT EXISTS %s_dm_em_ua_%s PARTITION OF %s
+				    FOR VALUES FROM (%L) TO (%L);', tableName, year, tableName, year, year+1);
+	RAISE NOTICE '%', cmdStmt;
+	IF executeFlag = 'Y' THEN
+		EXECUTE cmdStmt;
+		RAISE NOTICE '-- executed --';
+	END IF;
+
+-------------------------------------------------------------------------------------------------------------------

```

Comment:
For future reference, the rest of the tables occur in the script in alphabetical order, which makes them easier to find and confirm that they are in the script.  It is worth noting and keeping alphabetization when it already exists.